### PR TITLE
add missing 'return' keyword in atan2._eval_evalf. closes #17461

### LINKED ISSUE #17461: atan2._eval_evalf is missing a return keyword
`sympy.functions.atan2._eval_evalf` is missing a return keyword, and consequently `atan2` does not work with symbolic arguments that redefine their `_eval_evalf` method.

See https://groups.google.com/forum/#!msg/sympy/fpRIf9c79zw/taAssuspAgAJ

Solution: Add the return keyword.

```
import sympy
import sympy.functions as syf

class A(sympy.Symbol):
    def _eval_evalf(self, prec):
        return sympy.Float(5.0)

x = A('X', real=True)
y = A('Y', real=True)
print(syf.atan2(y, x).evalf())  # prints "atan2(Y, X)" instead of "0.785398163397448"
```

### PR DESCRIPTION
**Pull request by Geektrovert**
#### References to other Issues or PRs

Fixes #17461

#### Brief description of what is fixed or changed

Add missing 'return' keyword in sympy.functions.atan2._eval_evalf

#### Release Notes

* functions
  * Added missing 'return' keyword in sympy.functions.atan2._eval_evalf
  * Added test for sympy.functions.atan2._eval_evalf

### CODE REVIEW

**Comment by jksuom** [comment]
Can you add a test?

**Comment by oscargus** [comment]
There is a test in https://github.com/sympy/sympy/pull/17278/commits/8327b1bd008488c0b8c0b9893dbd0ad941f8d1c1 that you can use. (I reverted that commit once I saw this.)

**Comment by oscargus** [comment]
It may be that you'll have to add `is_extended_real = True` to the class to get it to work. My fix was slightly different (but I also think incorrect...).

File: sympy/functions/elementary/tests/test_trigonometric.py [hunk 1]
```diff
@@ -1068,6 +1068,15 @@ def test_atan2():
     raises(ArgumentIndexError, lambda: atan2(x, y).fdiff(3))
 
 
+def test_issue_17461():
+    class A(Symbol):
+        def _eval_evalf(self, prec):
```
Comment (oscargus) [inline sympy/functions/elementary/tests/test_trigonometric.py]:
```suggestion
        is_extended_real = True
        def _eval_evalf(self, prec):
```

**Comment by oscargus** [comment]
Thanks!

**Pull request merged**